src/Init/NotationExtra.lean

```diff
@@ -93,6 +93,45 @@ macro "Σ'" xs:explicitBinders ", " b:term : term => expandExplicitBinders `PSig
 macro:35 xs:bracketedExplicitBinders " × " b:term:35  : term => expandBrackedBinders `Sigma xs b
 macro:35 xs:bracketedExplicitBinders " ×' " b:term:35 : term => expandBrackedBinders `PSigma xs b
 
+@[appUnexpander Unit.unit] def unexpandUnit : Lean.PrettyPrinter.Unexpander
+  | `(Unit.unit) => `(())
+  | _            => throw ()
+
+@[appUnexpander List.nil] def unexpandListNil : Lean.PrettyPrinter.Unexpander
+  | `(List.nil) => `([])
```

Comment:
@dselsam We talked a bit about brittleness of unexpanders last time, and this is one simple example: since how identifiers should be compared depends on the use case (should they be interpreted as local or global identifiers?), we default to the most restrictive version, literal matching. So this unexpander should break after `open List`. Fortunately, this one can easily be avoided by replacing the constants with `$(_)`.
There has been some other thread on this and there were suggestions like substituting the full name before calling the unexpander, but I haven't seen a satisfying solution yet. It would be bad if matching identifiers in root function position magically worked in unexpanders, but not e.g. in argument position.

Reply:
Good catch. In fact, even the ones with atomic names are problematic on `ppAnalyze` branch, since we now sometimes add `_root_` prefixes. I changed all of the examples in `Init/NotationExtra.lean` to use `$(_)`.